mlflow/store/artifact/azure_blob_artifact_repo.py

```diff
@@ -38,10 +43,25 @@ def __init__(self, artifact_uri, client=None):
             self.client = BlobServiceClient(
                 account_url=account_url, credential=os.environ.get("AZURE_STORAGE_ACCESS_KEY")
             )
+        elif (
+            "AZURE_TENANT_ID" in os.environ
+            and "AZURE_CLIENT_ID" in os.environ
+            and "AZURE_CLIENT_SECRET" in os.environ
+        ):
+            from azure.identity import ClientSecretCredential
```

Comment:
this is certainly a welcome extension to the functionality here! If you're already pulling in azure-identity, is there any reason you chose not to use [the DefaultAzureCredential ](https://github.com/Azure/azure-sdk-for-python/tree/azure-storage-blob_12.7.0/sdk/identity/azure-identity#defaultazurecredential)to expand authentication types to include managed identities?

Reply:
Great suggestion! I've updated the PR to use the DefaultAzureCredential use instead of the ClientSecretCredential.